lib/actions/EndpointBuildApiGateway.js

```diff
@@ -841,20 +842,23 @@ module.exports = function(S) {
       _this.endpoint.path = _this.endpoint.path.join('/');
 
       // Create new access policy statement
-      let params          = {};
-      params.Action       = 'lambda:InvokeFunction';
-      params.FunctionName = _this.deployedLambda.FunctionArn;
-      params.Principal    = 'apigateway.amazonaws.com';
-      params.StatementId  = _this.lambdaPolicyStatementId;
-      params.SourceArn    = 'arn:aws:execute-api:'
-        + _this.evt.options.region
-        + ':'
-        + _this.awsAccountNumber
-        + ':'
-        +  _this.restApi.id
-        + '/*/'
-        + _this.endpoint.method
-        + _this.endpoint.path;
+      let params = {

```

Comment:
The code original code is a hot mess when it comes to mixing ES5 and ES6 and the use of parameters. Note that specifically `params` is reused multiple times throughout the code. However, my intention isn't to rewrite sls, but to fix some basic bugs. This seems to be much safer than starting to rewrite it (which would certainly be worth it).